src/legacy/core_plugins/visualizations/public/np_ready/public/vis_impl.js

```diff
@@ -62,6 +62,23 @@ class VisImpl extends EventEmitter {
     };
   }
 
+  initializeDefaultsFromSchemas(configStates, schemas) {
+    // Set the defaults for any schema which has them. If the defaults
+    // for some reason has more then the max only set the max number
+    // of defaults (not sure why a someone define more...
+    // but whatever). Also if a schema.name is already set then don't
+    // set anything.
+    const newConfigs = [...configStates];
+    schemas
+      .filter(schema => Array.isArray(schema.defaults) && schema.defaults.length > 0)
+      .filter(schema => !configStates.find(agg => agg.schema && agg.schema === schema.name))
+      .forEach(schema => {
+        const defaults = schema.defaults.slice(0, schema.max);
+        defaults.forEach(d => newConfigs.push(d));
+      });
+    return newConfigs;
+  }
```

Comment:
We should make a follow up task to go back and add some basic unit tests for `vis_impl` -- this method is something that might be nice to have a couple of tests for.